lua/lspconfig/configs.lua

```diff
@@ -113,6 +117,10 @@ function configs.__newindex(t, config_name, config_def)
         pcall(config.on_new_config, new_config, _root_dir)
       end
 
+      -- Reset values that have been changed from outside
+      -- Used by 'workspace/configuration' handler
+      new_config.settings = M.settings
```

Comment:
Won't this overwrite any settings that are set in on_new_config? This is the one thing I'm a bit concerned about

Reply:
Commented on wrong line, but I think you know what I meant lol

Reply:
Thanks for the review.
Oh, that's right, the value set in on_new_config will be overwritten... I just realized that.
Now I realize that it is better to update settings externally by doing something like `setup({on_new_config = nlspsettings.on_new_config})`.

Reply:
I think this pull request is incorrect, so can I close it?